spec/Section 5 -- Validation.md

```diff
@@ -1517,6 +1528,125 @@ query ($foo: Boolean = true, $bar: Boolean = false) {
 }
 ```
 
+### Defer And Stream Directives Are Used On Valid Root Field
+
+** Formal Specification **
+
+- For every {directive} in a document.
+- Let {directiveName} be the name of {directive}.
+- Let {mutationType} be the root Mutation type in {schema}.
+- Let {subscriptionType} be the root Subscription type in {schema}.
+- If {directiveName} is "defer" or "stream":
+  - The parent type of {directive} must not be {mutationType} or
+    {subscriptionType}.
+
+**Explanatory Text**
+
+The defer and stream directives are not allowed to be used on root fields of the
+mutation or subscription type.
+
+For example, the following document will not pass validation because `@defer`
+has been used on a root mutation field:
+
+```raw graphql counter-example
+mutation {
+  ... @defer {
+    mutationField
+  }
+}
+```
+
+### Defer And Stream Directive Labels Are Unique
+
+** Formal Specification **
+
+- For every {directive} in a document.
+- Let {directiveName} be the name of {directive}.
+- If {directiveName} is "defer" or "stream":
+  - For every {argument} in {directive}:
+    - Let {argumentName} be the name of {argument}.
+    - Let {argumentValue} be the value passed to {argument}.
+    - If {argumentName} is "label":
+      - {argumentValue} must not be a variable.
```

Comment:
that's a high impact thing, it needs change to parser. The parser now has to support the rule 'some values may not come from variables', and also identify defer's label arg as such variable. I would strongly vote for changing this to allow variables, and to run 'unique' rule at pre-execution time, against actual values of the label arg, maybe coming from variables.

Reply:
I'm not sure why you think it needs a change to the parser rather than to the validation stage as proposed here?

Reply:
Well, maybe change-to-parser is overkill, it depends on parser; for my parser in NGraphQL it would need adjustments I think. But maybe it's not for everyone. 
Still, it seems like a quite strange restriction, string literals only. I believe we don't have thing like this anywhere else. 
 Why not allow variables? - it might be beneficial in some cases, when labels are generated dynamically. The validation is then performed on actual values provided with the query, thru literals or variables, on pre-execution phase. The effect is the same (we catch dupes), but without this unusual restriction.

Reply:
Discussion topic is here: https://github.com/robrichard/defer-stream-wg/discussions/35

A lot of decisions have been to err on the side of stricter validation. If this does end up blocking real use cases a later proposal can explore loosening this restriction.